typings/resin-sdk.d.ts

```diff
@@ -857,12 +880,81 @@ declare namespace ResinSdk {
 					): Promise<void>;
 					remove(uuidOrId: string | number, tagKey: string): Promise<void>;
 				};
+				configVar: {
+					getAllByDevice(
+						uuidOrId: string | number,
+						options?: PineOptionsFor<DeviceConfigVariable>,
+					): Promise<DeviceConfigVariable[]>;
+					set(
+						uuidOrId: string | number,
+						key: string,
+						value: string,
+					): Promise<void>;
+					get(
+						uuidOrId: string | number,
+						key: string,
+					): Promise<string | undefined>;
+					remove(
+						uuidOrId: string | number,
+						key: string,
+					): Promise<string | undefined>;
+				};
+				envVar: {
+					getAllByDevice(
+						uuidOrId: string | number,
+						options?: PineOptionsFor<DeviceEnvironmentVariable>,
+					): Promise<DeviceEnvironmentVariable[]>;
+					set(
+						uuidOrId: string | number,
+						key: string,
+						value: string,
+					): Promise<void>;
+					get(
+						uuidOrId: string | number,
+						key: string,
+					): Promise<string | undefined>;
+					remove(
+						uuidOrId: string | number,
+						key: string,
+					): Promise<string | undefined>;
+				};
+				serviceVar: {
+					getAllByDevice(
+						uuidOrId: string | number,
+						options?: PineOptionsFor<DeviceServiceEnvironmentVariable>,
+					): Promise<DeviceServiceEnvironmentVariable[]>;
+					set(
+						uuidOrId: string | number,
+						serviceId: number,
+						key: string,
+						value: string,
+					): Promise<void>;
+					get(
+						uuidOrId: string | number,
+						serviceId: number,
+						key: string,
+					): Promise<string | undefined>;
+					remove(
+						uuidOrId: string | number,
+						serviceId: number,
+						key: string,
+					): Promise<string | undefined>;
+				};
 			};
 			service: {
 				getAllByApplication(
 					nameOrId: string | number,
 					options?: PineOptionsFor<Service>,
 				): Promise<Service[]>;
+				var: {
```

Comment:
That line of thought is tricky, because it's also true for devices. And then logs actually depend on devices, so should live within devices within applications... Etc etc etc. You're not wrong, but this namespaces here aren't supposed to directly represent the full hierarchy - they're supposed to make it clear & easy to find things.

It's totally debatable, but imo services are a sufficiently 'top level' frequently used resource that they should be directly accessible at the top here.